sabnzbd/downloader.py

```diff
@@ -654,27 +657,28 @@ def run(self):
             if readkeys:
                 read, _, _ = select.select(readkeys, (), (), 1.0)
 
-                # Add a sleep if there are too few results compared to the number of active connections
-                if self.can_be_slowed and len(read) < 1 + len(readkeys) / 10:
-                    time.sleep(self.sleep_time)
-
-                # Need to initialize the check during first 20 seconds
-                if self.can_be_slowed is None or self.can_be_slowed_timer:
-                    # Wait for stable speed to start testing
-                    if not self.can_be_slowed_timer and sabnzbd.BPSMeter.get_stable_speed(timespan=10):
-                        self.can_be_slowed_timer = time.time()
-
-                    # Check 10 seconds after enabling slowdown
-                    if self.can_be_slowed_timer and time.time() > self.can_be_slowed_timer + 10:
-                        # Now let's check if it was stable in the last 10 seconds
-                        self.can_be_slowed = sabnzbd.BPSMeter.get_stable_speed(timespan=10)
-                        self.can_be_slowed_timer = 0
-                        logging.debug("Downloader-slowdown: %r", self.can_be_slowed)
+                if self.can_be_slowed != 0:
```

Comment:
I don't understand why this is needed? 
Below is `self.can_be_slowed = BPSMeter.get_stable_speed(timespan=10)`, so it triggers the slow-down after 10 seconds of stable speed. After that it sets the timer `self.can_be_slowed_timer` while the slowdown is active and measures again for 10 seconds.

Reply:
It has also worked like this since 2.x so I'm pretty sure it's working 😄

Reply:
In the develop code, the sleep block is only entered if `self.can_be_slowed` is true. It can only be set true by https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/downloader.py#L654, but that block also sets `self.can_be_slowed_timer = 0`. The initialize block is only entered if `self.can_be_slowed is None` or `self.can_be_slowed_timer` is true, so as I under stand it, that `if` can't be true at the same time as `self.can_be_slowed` is true.

Reply:
The point of my refactor was to make it easier to read and not do more checks than necessary for all outcomes. If it is disabled, my code only needs to do one test. The current code does 3. (`if self.can_be_slowed` and `if self.can_be_slowed is None or self.can_be_slowed_timer`)